packages/gatsby/lib/schema/__tests__/infer-graphql-input-type.js

```diff
@@ -24,6 +24,10 @@ describe(`GraphQL Input args`, () => {
         title: `The world of dash and adventure`,
         blue: 100,
       },
+      anObjectArray: [
+        { aString: `some string`, aNumber: 2, aBoolean: true },
+        { aString: `some string`, aNumber: 2, anArray: [1, 2] },
```

Comment:
Good question. I do think properties that hold an array should be able to be empty or not, but for starters I think it would be ok if objects always have to have the same keys. And if the data doesn't some out of the CMS like that it's still easy to fix in client code.